### ISSUE TITLE
Sounds get distorted/skip/static in Windows while window is in background or resizing (new in 1.9.6)

### ISSUE THREAD

**Issue by davebalmer** [defold/defold#9966]
Sounds get interrupted and distorted when you ALT+TAB or resize game window. This is new, and so far just tested on Window build and run from editor.

**Steps to reproduce the behavior:**

1. run (debug or build) a game with a sound file (music) in its init (I am using an ogg file for example)
2. `ALT+TAB` back to the editor
3. move your mouse around in the editor window and interact with things
4. the sound playback skips and get corrupted (watch out for your speakers)

**Expected behavior (REQUIRED):**

The sound should play uninterrupted in the background as it did before the 1.9.6 release.

**Defold version (REQUIRED):**
  - Version 1.9.6

**Platforms (REQUIRED):**
 - Windows 10.x

**Minimal repro case project (OPTIONAL):**
Please attach a minimal project where the bug is reproduced. This will greatly help the person trying to investigate and fix the bug.

[defold-sound-test.zip](https://github.com/user-attachments/files/18204625/defold-sound-test.zip)

**Logs (OPTIONAL):**
Please provide relevant logs from engine, editor or build server. Learn where the logs are stored [here](https://defold.com/manuals/getting-help/#log-files).

**Workaround (OPTIONAL):**
None.

**Screenshots (OPTIONAL):**
N/A

**Additional context (OPTIONAL):**
None.

**Comment by davebalmer** [comment]
Tried with a build to Windows EXE and it did the same. Doesn't seem editor related, but window focus or not.

**Comment by JCash** [comment]
I couldn't reproduce the issue on my setup.
Could you provide the log from the sound system (`INFO:SOUND`) that's output from the engine when the game starts?

**Comment by davebalmer** [comment]
Sure, I'll look at it this evening.

**Comment by davebalmer** [comment]
```
INFO:SOUND: Mix format:
INFO:SOUND:   wFormatTag:       fffe  IEEE_FLOAT/PCM/EXTENSIBLE: 3 / 1 / fffe
INFO:SOUND:   nChannels:        2
INFO:SOUND:   nSamplesPerSec:   48000
INFO:SOUND:   nAvgBytesPerSec:  384000
INFO:SOUND:   nBlockAlign:      8
INFO:SOUND:   wBitsPerSample:   32
INFO:SOUND:   -> Format:        3  IEEE_FLOAT/PCM/EXTENSIBLE: 3 / 1 / fffe
```

I noticed when re-trying it that `ALT+TAB` triggered it the most. Once it started glitching, it got worse. About 15sec of fiddling to get it really bad.

```
Edition         Windows 10 Pro
Version         22H2
Processor       Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz   2.71 GHz
Installed RAM   8.00 GB
System type     64-bit operating system, x64-based processor
```

This machine is a bit of a potato, but never had this issue prior to Defold 1.9.6

**Comment by JCash** [comment]
Thanks fo rthe info!
2 channels is what I'd expect to work out-of-the-box :/

Yeah, in 1.9.6 we switched to the WASAPI backend, so there are likely to have some new bugs.

It's annoying that I can't reproduce it though. I'll have to mull it over some more...

**Comment by davebalmer** [comment]
This is still an issue. I tested on my other Windows test machine.

Windows 11 64 bit 23H2
Intel(R) Core(TM) i5-9400F CPU @ 2.90GHz   2.90 GHz
16.0 GB RAM

Defold startup debug:

```INFO:SOUND: Mix format:
INFO:SOUND:   wFormatTag:       fffe  IEEE_FLOAT/PCM/EXTENSIBLE: 3 / 1 / fffe
INFO:SOUND:   nChannels:        2
INFO:SOUND:   nSamplesPerSec:   48000
INFO:SOUND:   nAvgBytesPerSec:  384000
INFO:SOUND:   nBlockAlign:      8
INFO:SOUND:   wBitsPerSample:   32
INFO:SOUND:   -> Format:        3  IEEE_FLOAT/PCM/EXTENSIBLE: 3 / 1 / fffe
INFO:ENGINE: Initialised sound device 'default'
```

I noticed it was more pronounced when using my headphones through the FFINE usb mic than I did through the normal desktop speakers (also USB, Creative blob things).

This was most pronounced when ALT-TABing to other windows (eg. Chrome). BZZT! Crackle! lag lag Crackle! From editor and run from built EXE.

**Comment by britzl** [comment]
> I tested on my other Windows test machine.

To confirm: You tested with Defold 1.9.7 beta?

**Comment by davebalmer** [comment]
No, 1.9.6 just on another Windows machine. Is there a fix in the beta? If so I'll go figure out how to get it (gulp) and re-test. 😊

**Comment by britzl** [comment]
There are a few sound related fixed in 1.9.7 (link to the beta is in the release notes):

https://forum.defold.com/t/defold-1-9-7-beta/79429

**Comment by davebalmer** [comment]
OK I just tested 1.9.7 beta on the windows laptop. If anything, it seems **worse**. Playback on the `.ogg` loop stutters more than 1.9.6 when I switch windows. But less static? But the stutters are worse and more frequent (like it freezes for a half second every second or two, depending on if I'm doing anything in the other window like scrolling or typing).

I don't want to be stuck in 1.9.5. I guess I can do a workaround for now that makes the game mute sound when it doesn't have focus. But this isn't ideal. But I can't be the only one affected by this.

**Comment by davebalmer** [comment]
Wish I knew anything about sound internals in Windows and that my C chops weren't from the late 80's 😅 ...

**Comment by JCash** [comment]
I'll have a look tomorrow to see if I can output some more diagnostics about the audio device.
I really wish I could reproduce it, I'll try to do more work in the other applications, to see if that triggers anything.

**Comment by davebalmer** [comment]
Both my Windows machines have this issue, but the laptop I'm using is cheap used on ebay for US$100-150 😁 I'm sure it'd run your dev environment, though the compile time probably isn't great.

If that's an option, search for `MS Surface Laptop 1st Gen`. I'd ship you this one if I weren't using it. It's my EDC because it's so light and doesn't fry my legs while I'm working.

**Comment by davebalmer** [comment]
Wow changing the "Spatial Sound" Windows 10 option while my game is running slows the game down to like 1fps and makes it unresponsive...! Just noticed.

**Comment by AGulev** [comment]
Is it still the issue in the latest version of Defold?

**Comment by davebalmer** [comment]
Sadly yes. This happens on two different machines for me, one w Windows 10 and the with 11. One is a gaming desktop the other is a potato laptop.

**Comment by davebalmer** [comment]
This is ogg. I'll try wav and opus next and let you know.

**Comment by davebalmer** [comment]
Same problem with wav. Trying to load an opus file crashed.
